results/telemetry_db.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+require_once 'idObfuscation.php';
+
+define('TELEMETRY_SETTINGS_FILE', 'telemetry_settings.php');
+
+/**
+ * @return PDO|false
+ */
+function getPdo()
+{
+    if (
+        !file_exists(TELEMETRY_SETTINGS_FILE)
+        || !is_readable(TELEMETRY_SETTINGS_FILE)
+    ) {
+        return false;
+    }
+
+    require TELEMETRY_SETTINGS_FILE;
+
+    if (!isset($db_type)) {
+        return false;
+    }
+
+    $pdoOptions = [
+        PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION
+    ];
+
+    try {
+        if ('mysql' === $db_type) {
+            if (!isset(
+                $MySql_hostname,
+                $MySql_port,
+                $MySql_databasename,
+                $MySql_username,
+                $MySql_password
+            )) {
+                return false;
+            }
+
+            $dsn = 'mysql:'
```

Comment:
Why are you doing it this way? Any advantages over the old version?

Reply:
@adolfintel the advantage is that now all three database types are using the same way (PDO) for interacting with the DB. Postgresql and Sqlite where already using PDO, I just adjusted Mysql to also use PDO. So now only the instantiation of the PDO object is different, the execution of the actual queries now uses the same code for all three database types.

Reply:
Seeing your other comment about "that dsn thing which is new to me", the DSN is just a string which contains all the information that PDO needs to connect to the database (see https://www.php.net/manual/pdo.construct.php). Before these changes, Postgresql and Sqlite where already using such a DSN when instantiating PDO, it was just not named explititly like that in the code.

Reply:
Ok, thank you for clarifying that.
I'm mostly worried about the fact that you're appending the parameters into a string: what if the password contains a semicolon?

Reply:
@adolfintel good point, thanks :)

adjusted in 66fd64de7f2c7cd36dcc598010784598637a6610